src/Compilers/Core/Portable/Symbols/Attributes/CommonAttributeData.cs

```diff
@@ -292,9 +293,37 @@ private ObsoleteAttributeData DecodeObsoleteAttribute()
                 }
             }
 
-            return new ObsoleteAttributeData(ObsoleteAttributeKind.Obsolete, message, isError);
+            string? diagnosticId = null;
+            string? urlFormat = null;
+            foreach (var (name, value) in this.CommonNamedArguments)
+            {
+                if (diagnosticId is null && name == "DiagnosticId" && IsStringProperty("DiagnosticId"))
+                {
+                    diagnosticId = value.ValueInternal as string;
+                }
+                else if (urlFormat is null && name == "UrlFormat" && IsStringProperty("UrlFormat"))
```

Comment:
The strings `"DiagnosticId"` and `"UrlFormat"` are used multiple times in the PR. Is this worth adding a constant for or not really?

Reply:
I added constants to ObsoleteAttributeData just because the same literal strings are each used in 3 places.